src/app/core/core.state.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ActionReducerMap,
+  MetaReducer,
+  createFeatureSelector
+} from '@ngrx/store';
+import { storeFreeze } from 'ngrx-store-freeze';
+
+import { environment } from '@env/environment';
+
+import { initStateFromLocalStorage } from './meta-reducers/init-state-from-local-storage.reducer';
+import { debug } from './meta-reducers/debug.reducer';
+import { AuthState } from './auth/auth.models';
+import { authReducer } from './auth/auth.reducer';
+
+export const reducers: ActionReducerMap<AppState> = {
+  auth: authReducer
+};
+
+export const metaReducers: MetaReducer<AppState>[] = [
+  initStateFromLocalStorage
+];
+if (!environment.production) {
+  metaReducers.unshift(storeFreeze);
+  if (!environment.test) {
+    metaReducers.unshift(debug);
+  }
+}
+
+export const selectAuthState = createFeatureSelector<AppState, AuthState>(
```

Comment:
Currently we have only `AuthState` in here but in theory there can be more reducers and related states. This would be then more like `selectCoreState` which has auth state in it similarly like `selectExampleState` has `todos` and `stocks` in it.

Reply:
We can also thing about moving settings into core. The setting was implemented as a stand alone module with component, reducer, etc but is eagerly loaded so that's the question then. Does every eagerly loaded module have to be in core or can they be extracted away. Maybe it only has to be in core if any other core stuff needs its functionality. If not it can be extracted into stand alone eager module like it is now.

Reply:
Exactly what you're saying.
> Maybe it only has to be in core if any other core stuff needs its functionality. If not it can be extracted into stand alone eager module like it is now.

Reply:
Yeah, let's go with that. Still, the previous comment talks about `selectCoreState ` vs `selectAuthState`, please have a look and let me know what do you think.

Reply:
I think `selectAuthState` is better in this case, we are selecting `auth` from the `AppState`.
If for example settings would also be part of the `AppState` we would have another selector for settings `export const selectSettingState = createFeatureSelector<AppState, SettingsState>( 'settings');`

Reply:
does that make sense what I told here? Or should we go for `selectCoreState`?

Reply:
@timdeschryver yes sure! I just was a bit tight on time last couple of days cuz I am visiting family in Slovakia :) Will prolly merge now as I already did for `lint-staged` :)

Reply:
Sorry, wasn't meant to be a "Hey merge me" message 😅 
Enjoy your time in Slovakia!

Reply:
No worries, was not taken like that :) Thanks for the great PR and discussion about the concepts!